abci/types/types.proto

```diff
@@ -254,6 +254,16 @@ message Event {
       [(gogoproto.nullable) = false, (gogoproto.jsontag) = "attributes,omitempty"];
 }
 
+// TxResult contains results of executing the transaction.
+//
+// One usage is indexing transaction results.
+message TxResult {
+  int64             height = 1;
+  uint32            Index  = 2;
+  bytes             tx     = 3;
+  ResponseDeliverTx result = 4 [(gogoproto.nullable) = false];
```

Comment:
Is it necessary to set `gogoproto.nullable`? I thought Protobuf 3 removed all support for `null` values, so it's a bit unclear to me what purpose this option serves.

Reply:
you are correct about null in proto but the go generated type will be a ptr if it's using another prototype, so to remove the ptr we are doing this

Reply:
Huh, interesting - surprising to me that structs would be generated as pointers by default, since both Protobuf and Go makes such a big deal out of zero values.

Reply:
here it is stated in the docs, https://developers.google.com/protocol-buffers/docs/reference/go-generated#singular-message

then we use gogoslick which reduces the number of pointers but still need it in places
```
protoc-gen-gogofast (same as gofast, but imports gogoprotobuf)
protoc-gen-gogofaster (same as gogofast, without XXX_unrecognized, less pointer fields)
protoc-gen-gogoslick (same as gogofaster, but with generated string, gostring and equal methods)
```